test/Microsoft.AspNet.Mvc.FunctionalTests/MvcTestFixture.cs

```diff
@@ -2,62 +2,105 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
+using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Net.Http;
 using System.Reflection;
-using System.Runtime.Versioning;
 using Microsoft.AspNet.Builder;
 using Microsoft.AspNet.Hosting;
 using Microsoft.AspNet.Mvc.Actions;
 using Microsoft.AspNet.TestHost;
+using Microsoft.AspNet.Testing;
 using Microsoft.Dnx.Runtime;
 using Microsoft.Dnx.Runtime.Infrastructure;
 using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.Logging.Testing;
 
 namespace Microsoft.AspNet.Mvc.FunctionalTests
 {
-    public class MvcFixture : IDisposable
+    public class MvcTestFixture : IDisposable
     {
-        public MvcFixture(object startupInstance)
+        private readonly TestServer _server;
+
+        public MvcTestFixture(object startupInstance)
         {
             var startupTypeInfo = startupInstance.GetType().GetTypeInfo();
-            var configureMethod = (Action<IApplicationBuilder>)startupTypeInfo
+            var configureApplication = (Action<IApplicationBuilder>)startupTypeInfo
                 .DeclaredMethods
-                .First(m => m.Name == "Configure")
-                .CreateDelegate(typeof(Action<IApplicationBuilder>), startupInstance);
+                .FirstOrDefault(m => m.Name == "Configure" && m.GetParameters().Length == 1)
+                ?.CreateDelegate(typeof(Action<IApplicationBuilder>), startupInstance);
+            if (configureApplication == null)
+            {
+                var configureWithLogger = (Action<IApplicationBuilder, ILoggerFactory>)startupTypeInfo
+                    .DeclaredMethods
+                    .FirstOrDefault(m => m.Name == "Configure" && m.GetParameters().Length == 2)
+                    ?.CreateDelegate(typeof(Action<IApplicationBuilder, ILoggerFactory>), startupInstance);
+                Debug.Assert(configureWithLogger != null);
+
+                configureApplication = application => configureWithLogger(application, NullLoggerFactory.Instance);
```

Comment:
If all we do is pass in a `NullLoggerFactory`, couldn't we simply change these websites to not use this `Configure` method?